insonmnia/dwh/sql.go

```diff
@@ -623,73 +639,78 @@ func (c *sqlStorage) GetDealChangeRequestsByID(conn queryConn, changeRequestID *
 }
 
 func (c *sqlStorage) InsertDealCondition(conn queryConn, condition *pb.DealCondition) error {
-	_, err := conn.Exec(c.commands.insertDealCondition,
-		condition.SupplierID.Unwrap().Hex(),
-		condition.ConsumerID.Unwrap().Hex(),
-		condition.MasterID.Unwrap().Hex(),
-		condition.Duration,
-		condition.Price.PaddedString(),
-		condition.StartTime.Seconds,
-		condition.EndTime.Seconds,
-		condition.TotalPayout.PaddedString(),
-		condition.DealID.Unwrap().String())
+	query, args, err := c.builder().Insert("DealConditions").Columns(c.tablesInfo.DealConditionColumns[1:]...).
+		Values(
+			condition.SupplierID.Unwrap().Hex(),
+			condition.ConsumerID.Unwrap().Hex(),
+			condition.MasterID.Unwrap().Hex(),
+			condition.Duration,
+			condition.Price.PaddedString(),
+			condition.StartTime.Seconds,
+			condition.EndTime.Seconds,
+			condition.TotalPayout.PaddedString(),
+			condition.DealID.Unwrap().String(),
+		).ToSql()
+	_, err = conn.Exec(query, args...)
 	return err
 }
 
 func (c *sqlStorage) UpdateDealConditionPayout(conn queryConn, dealConditionID uint64, payout *big.Int) error {
-	_, err := conn.Exec(c.commands.updateDealConditionPayout, util.BigIntToPaddedString(payout), dealConditionID)
+	query, args, err := c.builder().Update("DealConditions").Set("TotalPayout", util.BigIntToPaddedString(payout)).
+		Where("Id = ?", dealConditionID).ToSql()
+	_, err = conn.Exec(query, args...)
 	return err
 }
 
 func (c *sqlStorage) UpdateDealConditionEndTime(conn queryConn, dealConditionID, eventTS uint64) error {
-	_, err := conn.Exec(c.commands.updateDealConditionEndTime, eventTS, dealConditionID)
+	query, args, err := c.builder().Update("DealConditions").Set("EndTime", eventTS).
+		Where("Id = ?", dealConditionID).ToSql()
+	_, err = conn.Exec(query, args...)
 	return err
 }
 
-func (c *sqlStorage) InsertDealPayment(conn queryConn, payment *pb.DealPayment) error {
```

Comment:
Just curious, what the reason to delete this function? It becomes irrelevant for us?

Reply:
it was, ahem, added after some analyst’s request, but proved to be useless